File: src/main/java/dev/retrotv/crypto/twe/des/DESECB.java
package dev.retrotv.crypto.twe.des;

import dev.retrotv.crypto.exception.KeyGenerateException;

import javax.crypto.KeyGenerator;
import java.security.Key;
import java.security.NoSuchAlgorithmException;

import static dev.retrotv.enums.CipherAlgorithm.DESECB;

public class DESECB extends DES {
    public DESECB() {
        this.algorithm = DESECB;
    }

    @Override
    public Key generateKey() throws KeyGenerateException {
        try {
            KeyGenerator keyGenerator = KeyGenerator.getInstance("DES");
            return keyGenerator.generateKey();
        } catch (NoSuchAlgorithmException e) {
            throw new KeyGenerateException("NoSuchAlgorithmException: \n지원하지 않는 암호화 알고리즘 입니다.");
        }
    }
}

File: src/main/kotlin/dev/retrotv/crypto/twe/des/DESECB.kt
package dev.retrotv.crypto.twe.des

import dev.retrotv.enums.Algorithm

/**
 * DES/ECB 양방향 암호화 클래스 입니다.
 *
 * @author  yjj8353
 * @since   1.0.0
 */
@Deprecated("해킹에 취약한 양방향 암호화 알고리즘 입니다.")
class DESECB : DES() {

    init {
        algorithm = Algorithm.Cipher.DESECB
    }
}
